pkg/util/bandwidth/utils.go

```diff
@@ -36,31 +36,53 @@ func validateBandwidthIsReasonable(rsrc *resource.Quantity) error {
 }
 
 // ExtractPodBandwidthResources extracts the ingress and egress from the given pod annotations
-func ExtractPodBandwidthResources(podAnnotations map[string]string) (ingress, egress *resource.Quantity, err error) {
+func ExtractPodBandwidthResources(podAnnotations map[string]string) (ingress, ingressBurst, egress, egressBurst *resource.Quantity, err error) {
 	if podAnnotations == nil {
-		return nil, nil, nil
+		return nil, nil, nil, nil, nil
 	}
 	str, found := podAnnotations["kubernetes.io/ingress-bandwidth"]
 	if found {
 		ingressValue, err := resource.ParseQuantity(str)
 		if err != nil {
-			return nil, nil, err
+			return nil, nil, nil, nil, err
 		}
 		ingress = &ingressValue
 		if err := validateBandwidthIsReasonable(ingress); err != nil {
-			return nil, nil, err
+			return nil, nil, nil, nil, err
+		}
+	}
+	str, found = podAnnotations["kubernetes.io/ingress-burst-bandwidth"]
+	if found {
+		ingressBurstValue, err := resource.ParseQuantity(str)
+		if err != nil {
+			return nil, nil, nil, nil, err
+		}
+		ingressBurst = &ingressBurstValue
+		if err := validateBandwidthIsReasonable(ingressBurst); err != nil {
+			return nil, nil, nil, nil, err
 		}
 	}
 	str, found = podAnnotations["kubernetes.io/egress-bandwidth"]
 	if found {
 		egressValue, err := resource.ParseQuantity(str)
 		if err != nil {
-			return nil, nil, err
+			return nil, nil, nil, nil, err
 		}
 		egress = &egressValue
 		if err := validateBandwidthIsReasonable(egress); err != nil {
-			return nil, nil, err
+			return nil, nil, nil, nil, err
+		}
+	}
+	str, found = podAnnotations["kubernetes.io/egress-burst-bandwidth"]
```

Comment:
These blocks look like 4 times the same (expect for the annotation) do you think it make sense to refactor this into a single method and add some unit tests?